### ISSUE TITLE
[@SerenityWithCucumber] Report Portal creates redundant launches for multi module project

### ISSUE THREAD

**Issue by nstbushneva** [Invictum/serenity-reportportal-integration#63]
On running tests in multimodule test-automation-framework, Report Portal creates a separate launch for each module. It generates redundant launches for modules which don't have any tests with the specified tag. 
![redundant-launch](https://user-images.githubusercontent.com/36731862/54091087-75a4f600-4384-11e9-8694-ef72cd2db601.gif)

Would it be possible to keep all tests results in one launch even though tests are defined in different maven modules? 

The project [simple-multimodule-project.zip](https://github.com/Invictum/serenity-reportportal-integration/files/2949995/simple-multimodule-project.zip):  
* serenity-reportportal-integration [1.3.0]
* maven-failsafe-plugin [2.22.1]
* serenity-core [2.0.20]
* serenity-cucumber [1.9.21]

Execute the following command to reproduce the issue: 
```bash
 mvn clean verify -Dcucumber.options="--tags '@API'"
```

**Comment by Invictum** [comment]
@nstbushneva 

jUnit is configured to run tests in parallel using JVM forking.
```
      <plugin>
        <artifactId>maven-failsafe-plugin</artifactId>
        <version>${maven.failsafe.plugin.version}</version>
        <configuration>
          <includes>
            <include>**/*Test.java</include>
          </includes>
          <parallel>classes</parallel>
          <threadCount>${threads.number}</threadCount>
          <forkCount>${threads.number}</forkCount>
          <perCoreThreadCount>false</perCoreThreadCount>
          <testFailureIgnore>false</testFailureIgnore>
        </configuration>
        <executions>
          <execution>
            <goals>
              <goal>integration-test</goal>
              <goal>verify</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
```
So tests on each submodule run on a different JVMs. But for now integration controls RP communication client on one JVM level, so practically with your example several client created, as a result you got several launches in RP.

Potentially is it possible to avoid it with cross JVM communication, but efforts to do it is too high comparing to this use case rarity.

**Comment by slavamarkovski** [comment]
Hello,

Do you have any plans to fix the issue? We faced with the same problem recently.
As a workaround it is possible to set unique tag for each set of runs and merge them on full run completion using RP API. 

Thanks.

**Comment by Invictum** [comment]
@slavamarkovski 
Your proposed workaround idea sounds good. I will try to apply it with little modification.
Thanks for contribution

**Comment by slavamarkovski** [comment]
@Invictum sorry for pushing but do you any rough ETA for this issue?

**Comment by Invictum** [comment]
@slavamarkovski 

I made a small investigation and actually it is more complex than expected. It requires cross JVM communication even with merge approach.

As for estimates: It depends on my free time, but I will try to implement it in a few weeks.

**Comment by Invictum** [comment]
Implemented. Will be available on 1.4.2+

@slavamarkovski 
Before release feature is available in snapshots, related documentation can be found in `README.md` in `development` branch
